pkg/trace/agent/tags.go

```diff
@@ -280,14 +284,22 @@ func NormalizeTag(tag string) string {
 			continue
 		}
 
-		c = unicode.ToLower(c)
+		if utf8.ValidRune(c) && unicode.IsUpper(c) {
+			// lowercase this character
+			if low := unicode.ToLower(c); utf8.RuneLen(c) == utf8.RuneLen(low) {
```

Comment:
Yeah I think this is a good compromise. Actually handling this for all cases would require a separate buffer whose removal was the whole point of this set of PRs.

Reply:
For future reference if anyone looks at this PR, here is a short algorithm showing all the 24 edge cases: https://play.golang.org/p/Af1bIusSbfq